price-feeder/oracle/provider/gate.go

```diff
@@ -308,28 +308,44 @@ func (p *GateProvider) messageReceived(messageType int, bz []byte) {
 		return
 	}
 
-	var gateEvent GateEvent
-	if err := json.Unmarshal(bz, &gateEvent); err != nil {
-		p.logger.Debug().Msg("received a message that is not an event")
-	} else {
+	var (
+		gateEvent GateEvent
+		gateErr   error
+		tickerErr error
+		candleErr error
+	)
+
+	gateErr = json.Unmarshal(bz, &gateEvent)
+	if gateErr == nil {
 		switch gateEvent.Result.Status {
 		case "success":
 			return
 		case "":
+			// msg was not an event, it will try to marshal to ticker message.
 			break
 		default:
 			p.reconnect()
+			return
 		}
 	}
 
-	if err := p.messageReceivedTickerPrice(bz); err != nil {
-		// msg is not a ticker, it will try to marshal to candle message.
-		p.logger.Debug().Err(err).Msg("unable to unmarshal ticker")
-	} else {
+	tickerErr = p.messageReceivedTickerPrice(bz)
+	if tickerErr == nil { // succeded to unmarshal ticker price
 		return
 	}
-	if err := p.messageReceivedCandle(bz); err != nil {
-		p.logger.Debug().Err(err).Msg("unable to unmarshal candle")
+	// msg is not a ticker, it will try to marshal to candle message.
+
+	candleErr = p.messageReceivedCandle(bz)
+	if candleErr == nil {
+		return
+	}
+
+	if tickerErr != nil || candleErr != nil || gateErr != nil {
```

Comment:
This if statement is guaranteed to be true

Reply:
Thanks, I did not notice that